### PR TITLE
Add unsubscribe page

### LINKED ISSUE #4: Make unsubscribe form
The path `/unsubscribe` exists, but it's essentially a static page. Make it a form for with a single field for the email address. The code for calling the Sparkpost API can be found in https://github.com/geowa4/elixir-sparkpost-experiements.

### PR DESCRIPTION
**Pull request by dantswain**
Closes #4 

Adds routes:
* `GET /unsubscribe?email=email` - Shows the unsubscribe form, fills out email field if supplied
* `DELETE /subscriptions` - Actually performs the unsubscribe

The unsubscribe is done asynchronously.  We do not retry if it fails.  This could be done later as a refinement, I think?

I considered having `GET /unsubscribe` redirect to `GET /subscriptions/unsubscribe`, but I don't know that that adds any real value.  The `/unsubscribe` url is nice and short.

### CODE REVIEW

File: app/lib/rocdev_web/templates/subscription/unsubscribe.html.eex [hunk 1]
```diff
@@ -1,3 +1,6 @@
-<div class="jumbotron">
-  <h2>Sorry to see you go.</h2>
-</div>
\ No newline at end of file
+Whatever.  See if we care.
```
Comment (geowa4) [inline app/lib/rocdev_web/templates/subscription/unsubscribe.html.eex]: Uh...

File: app/lib/rocdev_web/controllers/subscription_controller.ex [hunk 2]
```diff
@@ -1,7 +1,39 @@
...
   use RocdevWeb, :controller
 
-  def unsubscribe(conn, _params) do
-    render conn, "unsubscribe.html"
+  alias Rocdev.MailingList
+
+  # raises a good point:
+  # https://davidcel.is/posts/stop-validating-email-addresses-with-regex/
+  # ¯\_(ツ)_/¯
+  @email_regex ~r/.+@.+\..+/i
```
Comment (geowa4) [inline app/lib/rocdev_web/controllers/subscription_controller.ex]: I guess the owners of `http://to.` will never join this community.

**Review by geowa4 — CHANGES_REQUESTED** [review]
I appreciate the joke in unsubscribe.html, but I'm not sure everyone will. As for the email regex, can we make that `.+@.+`?

File: app/lib/rocdev_web/templates/subscription/unsubscribe.html.eex (hunk 1 above)
Comment (dantswain) [reply app/lib/rocdev_web/templates/subscription/unsubscribe.html.eex]: https://www.youtube.com/watch?v=SomQX54qhz0

**Review by geowa4 — APPROVED** [review]

**Pull request merged**